src/System.IO.FileSystem/tests/FileInfo/GetSetTimes.cs

```diff
@@ -99,6 +117,33 @@ public void CopyToMillisecondPresent()
             Assert.NotEqual(0, output.LastWriteTime.Millisecond);
         }
 
+        [ConditionalFact(nameof(isNotHFS))]
+        public void CopyToNanoSecondsPresent()
+        {
+            FileInfo input = GetNonZeroNanoSeconds();
+            FileInfo output = new FileInfo(Path.Combine(GetTestFilePath(), input.Name));
+
+            output.Directory.Create();
+            output = input.CopyTo(output.FullName, true);
+
+            Assert.Equal(input.LastWriteTime.Ticks, output.LastWriteTime.Ticks);
+            Assert.NotEqual(0, output.LastWriteTime.Ticks % 10);
+            Assert.NotEqual(0, input.LastWriteTime.Ticks % 10);
```

Comment:
Why is this assert needed here on the input?  GetNonZeroNanoSeconds should be guaranteed to return what it says it'll return; if it can't, it should be the one to throw / assert.